### PR TITLE
feat: [CDS-47337]: Create CSV report for Active services

### PR DESCRIPTION
**Pull request by ivan-83**
Added API for preparing and downloading CD active services CSV report. 

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/43154)

### CODE REVIEW

**Review by vladimir-peric — DISMISSED** [review]
> Reviewed 3 of 6 files at r1, 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43154)* status: 8 file(s) to review. 1 of 2 approvals obtained.

**Review by praveensugavanam — APPROVED** [review]
> Reviewed 3 of 6 files at r1, 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43154)* status: :shipit: complete! ✓

File: 945-license-usage-sdk/src/main/java/io/harness/licensing/usage/interfaces/LicenseUsageInterface.java [hunk 1]
```diff
@@ -29,4 +30,14 @@
...
+
+  /**
+   * Get license usage CSV report
+   *
+   * @param accountIdentifier the account identifier
+   * @param module the module
+   * @param currentTsInMs the current timestamp in ms
+   * @return CSV report file
+   */
+  File getLicenseUsageCSVReport(String accountIdentifier, ModuleType module, long currentTsInMs);
```
Comment (neelam-harness) [inline 945-license-usage-sdk/src/main/java/io/harness/licensing/usage/interfaces/LicenseUsageInterface.java:42]: nit: rename `module` -> `moduleType`

File: 125-cd-nextgen/src/main/java/io/harness/cdng/usage/mapper/ActiveServiceMapper.java [hunk 2]
```diff
@@ -62,4 +62,17 @@ public static ActiveServiceFetchData buildActiveServiceFetchData(
         .endTSInMs(timestamp)
         .build();
   }
+
+  public static ActiveServiceFetchData buildActiveServiceFetchData(
```
Comment (neelam-harness) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/usage/mapper/ActiveServiceMapper.java:66]: We have same method above, wondering if we can re-use it by passing down the optionals instead ? If we don't need certain attributes we can just pass down Optional.empty().

**Review by neelam-harness — CHANGES_REQUESTED** [review]

**Review by ivan-83 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43154)* status: 4 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/usage/mapper/ActiveServiceMapper.java` line 66 at r2](https://reviewable.io/reviews/harness/harness-core/43154#-NMna6eTAPoWs2nbrC8w-r2-66:-NN6_3AX3UnNyb7j7cHb:b-e8fmge) ([raw file](https://github.com/harness/harness-core/blob/f187a9702aa7353d61a6a5322f7dcc1a5f0d7a40/125-cd-nextgen/src/main/java/io/harness/cdng/usage/mapper/ActiveServiceMapper.java#L66)):*
<details><summary><i>Previously, neelam-harness wrote…</i></summary><blockquote>

We have same method above, wondering if we can re-use it by passing down the optionals instead ? If we don't need certain attributes we can just pass down Optional.empty().
</blockquote></details>

We avoid passing Optional as a method param because the number of checks increases, beside check whether the value is present or not we need to check whether the passed optional arg itself is null. 
___
*[`945-license-usage-sdk/src/main/java/io/harness/licensing/usage/interfaces/LicenseUsageInterface.java` line 42 at r2](https://reviewable.io/reviews/harness/harness-core/43154#-NMna6eTAPoWs2nbrC8x-r2-42:-NN6_--I7NwHh7xwJqbG:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/f187a9702aa7353d61a6a5322f7dcc1a5f0d7a40/945-license-usage-sdk/src/main/java/io/harness/licensing/usage/interfaces/LicenseUsageInterface.java#L42)):*
<details><summary><i>Previously, neelam-harness wrote…</i></summary><blockquote>

nit: rename `module` -> `moduleType`
</blockquote></details>

Done.

**Review by vladimir-peric — APPROVED** [review]
> Reviewed 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43154)* status: 4 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by deepak-harness — APPROVED** [review]
> Reviewed 2 of 6 files at r1, 2 of 5 files at r2, 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43154)* status: 1 change requests. 1. unresolved discussions.

**Review by praveensugavanam — APPROVED** [review]
> Reviewed 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43154)* status: 1 change requests. 1. unresolved discussions.

**Review by neelam-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43154)* status: :shipit: complete! ✓

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/usage/mapper/ActiveServiceMapper.java` line 66 at r2](https://reviewable.io/reviews/harness/harness-core/43154#-NMna6eTAPoWs2nbrC8w-r2-66:-NN8IRr61YEua1YPq9Bc:b-f4pg81) ([raw file](https://github.com/harness/harness-core/blob/f187a9702aa7353d61a6a5322f7dcc1a5f0d7a40/125-cd-nextgen/src/main/java/io/harness/cdng/usage/mapper/ActiveServiceMapper.java#L66)):*
<details><summary><i>Previously, ivan-83 (Ivan Mijailovic) wrote…</i></summary><blockquote>

We avoid passing Optional as a method param because the number of checks increases, beside check whether the value is present or not we need to check whether the passed optional arg itself is null. 
</blockquote></details>

Makes sense. The only issue I see with this model is that we will keep adding more and more methods to build customized ActiveServiceFetchData each time but I'll let you decide on this :)

**Review by ivan-83 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43154)* status: 1 change requests. 1. unresolved discussions. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/usage/mapper/ActiveServiceMapper.java` line 66 at r2](https://reviewable.io/reviews/harness/harness-core/43154#-NMna6eTAPoWs2nbrC8w-r2-66:-NN8PO8O8GpQ3_AjaOA6:bb77ufg) ([raw file](https://github.com/harness/harness-core/blob/f187a9702aa7353d61a6a5322f7dcc1a5f0d7a40/125-cd-nextgen/src/main/java/io/harness/cdng/usage/mapper/ActiveServiceMapper.java#L66)):*
<details><summary><i>Previously, neelam-harness wrote…</i></summary><blockquote>

Makes sense. The only issue I see with this model is that we will keep adding more and more methods to build customized ActiveServiceFetchData each time but I'll let you decide on this :)
</blockquote></details>

Yes.

**Review by neelam-harness — COMMENTED** [review]
> Reviewed 2 of 6 files at r1, 2 of 5 files at r2, 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43154)* status: 1 change requests. 1. unresolved discussions.

**Review by neelam-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43154)* status: :shipit: complete! ✓

**Pull request merged**